diff --git a/python/CMakeLists.txt b/python/CMakeLists.txt
index 5e76b5e..e4549f3 100644
--- a/python/CMakeLists.txt
+++ b/python/CMakeLists.txt
@@ -21,7 +21,6 @@ set(OPENMMPLUMED_HEADER_DIR "${CMAKE_SOURCE_DIR}/openmmapi/include")
 set(OPENMMPLUMED_LIBRARY_DIR "${CMAKE_BINARY_DIR}")
 configure_file(${CMAKE_CURRENT_SOURCE_DIR}/setup.py ${CMAKE_CURRENT_BINARY_DIR}/setup.py)
 add_custom_command(TARGET PythonInstall
-    COMMAND "${PYTHON_EXECUTABLE}" setup.py build
-    COMMAND "${PYTHON_EXECUTABLE}" setup.py install
+    COMMAND ${PYTHON_EXECUTABLE} -m pip install -v .
     WORKING_DIRECTORY "${CMAKE_CURRENT_BINARY_DIR}"
 )
diff --git a/python/plumedplugin.i b/python/plumedplugin.i
index 99b61ac..35408da 100644
--- a/python/plumedplugin.i
+++ b/python/plumedplugin.i
@@ -1,6 +1,6 @@
 %module openmmplumed
 
-
+%include "factory.i"
 %import(module="simtk.openmm") "swig/OpenMMSwigHeaders.i"
 %include "swig/typemaps.i"
 %include "std_string.i"
